internal/state.go

```diff
@@ -85,14 +85,37 @@ func (s *State) IdleWorkers() []Worker {
 
 // StrayInstances returns a list of instance IDs that don't have a corresponding
 // worker in the worker pool.
-func (s *State) StrayInstances() (out []string) {
+func (s *State) StrayInstances() []string {
+	var res []string
 	for instanceID := range s.inServiceInstanceIDs {
 		if _, ok := s.workersByInstanceID[instanceID]; !ok {
-			out = append(out, string(instanceID))
+			res = append(res, string(instanceID))
 		}
 	}
 
-	return
+	res = append(res, s.detachedNotTerminatedInstances()...)
```

Comment:
I just want to check something - do you know how [DescribeInstances](https://github.com/spacelift-io/ec2-workerpool-autoscaler/blob/41323ce32ca1e2eb89cee339857f4b40518270dc/internal/controller.go#L102) will behave if a particular instance ID doesn't exist?

What I'm concerned about is the opposite situation, where there's a delay in us marking a worker as gone in the Spacelift API, leading to that instance ID being returned from the API, then leading to the describe call failing.

Reply:
I think we should get the empty array of results, from docs:
> If you specify an instance that you do not own, it is not included in the output.

btw, interesting thing:
> Recently terminated instances might appear in the returned results. This interval is usually less than one hour.

Reply:
So maybe we should also check the `instanceState` property and filter out terminated instances? https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Instance.html

Reply:
> So maybe we should also check the instanceState property and filter out terminated instances? https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Instance.html

I think yes, because if we delete the stray instance once, the second rerun of the script will find the same instance(probably in the terminated state), and we will try to delete it again.

Reply:
Ok - as long as we're handling it somehow sounds good.